HopsanGUI/Dialogs/OptimizationScriptWizard.cpp

```diff
@@ -0,0 +1,1611 @@
+        pMinLineEdit->setValidator(new QDoubleValidator());
+        QLineEdit *pMaxLineEdit = new QLineEdit(max, this);
+        pMaxLineEdit->setValidator(new QDoubleValidator());
+        QToolButton *pRemoveButton = new QToolButton(this);
+        pRemoveButton->setIcon(QIcon(QString(ICONPATH) + "Hopsan-Discard.png"));
+        pRemoveButton->setToolTip("Remove Parameter");
+        connect(pRemoveButton, SIGNAL(clicked()), this, SLOT(removeParameter()));
+
+        mpParameterLabels.append(pLabel);
+        mpParameterMinLineEdits.append(pMinLineEdit);
+        mpParameterMaxLineEdits.append(pMaxLineEdit);
+        mpParameterRemoveButtons.append(pRemoveButton);
+
+        int row = mpParametersLayout->rowCount();
+        mpParametersLayout->addWidget(pMinLineEdit, row, 0);
+        mpParametersLayout->addWidget(pLabel, row, 1);
+        mpParametersLayout->addWidget(pMaxLineEdit, row, 2);
+        mpParametersLayout->addWidget(pRemoveButton, row, 3);
+    }
+    else
+    {
+        int i=0;
+        for(; i<mSelectedParameters.size(); ++i)
+        {
+            if(mSelectedComponents.at(i) == item->parent()->text(0) &&
+               mSelectedParameters.at(i) == item->text(0))
+            {
+                break;
+            }
+        }
+        mpParametersLayout->removeWidget(mpParameterLabels.at(i));
+        mpParametersLayout->removeWidget(mpParameterMinLineEdits.at(i));
+        mpParametersLayout->removeWidget(mpParameterMaxLineEdits.at(i));
+        mpParametersLayout->removeWidget(mpParameterRemoveButtons.at(i));
+
+        delete(mpParameterLabels.at(i));
+        delete(mpParameterMinLineEdits.at(i));
+        delete(mpParameterMaxLineEdits.at(i));
+        delete(mpParameterRemoveButtons.at(i));
+
+        mpParameterLabels.removeAt(i);
+        mpParameterMinLineEdits.removeAt(i);
+        mpParameterMaxLineEdits.removeAt(i);
+        mpParameterRemoveButtons.removeAt(i);
+
+        mSelectedParameters.removeAt(i);
+        mSelectedComponents.removeAt(i);
+    }
+}
+
+
+//! @brief Removes an objective function from the selected functions
+void OptimizationScriptWizard::removeParameter()
+{
+    QToolButton *button = qobject_cast<QToolButton *>(sender());
+    int i = mpParameterRemoveButtons.indexOf(button);
+
+    QTreeWidgetItem *selectedItem = findParameterTreeItem(mSelectedComponents.at(i), mSelectedParameters.at(i));
+    if(selectedItem)
+    {
+        selectedItem->setCheckState(0, Qt::Unchecked);
+    }
+    else
+    {
+
+    //Parameter is not in list (should not really happen), so remove it here instead
+    mpParametersLayout->removeWidget(mpParameterLabels.at(i));
+    mpParametersLayout->removeWidget(mpParameterMinLineEdits.at(i));
+    mpParametersLayout->removeWidget(mpParameterMaxLineEdits.at(i));
+    mpParametersLayout->removeWidget(mpParameterRemoveButtons.at(i));
+
+    delete(mpParameterLabels.at(i));
+    delete(mpParameterMinLineEdits.at(i));
```

Comment:
Here and in other places, maybe should use ->deleteLater(), in case events or signals are queued to these widgets